hedera-node/src/main/java/com/hedera/services/store/AccountStore.java

```diff
@@ -163,16 +150,15 @@ private Account loadEntityOrFailWith(Id id, @Nullable ResponseCodeEnum explicitR
 	/**
 	 * Persists the given account to the Swirlds state.
 	 *
-	 * @param account
-	 * 		the account to save
+	 * @param account the account to save
 	 */
-	public void persistAccount(Account account) {
+	public void commitAccount(Account account) {
 		final var id = account.getId();
-		final var key = EntityNum.fromLong(id.getNum());
 
-		final var mutableAccount = accounts.get().getForModify(key);
+		final var mutableAccount = accounts.getImmutableRef(id.asGrpcAccount());
```

Comment:
Here we want `accounts.getRef()`, so that behind the scenes we call `MerkleMap.getForModify()`